services/s3/src/test/java/software/amazon/awssdk/services/s3/utils/ChecksumUtils.java

```diff
@@ -66,4 +75,24 @@ private static MessageDigest createMessageDigest() {
             throw new RuntimeException("Unable to create SHA-256 MessageDigest instance", e);
         }
     }
+
+    public static File fixedLengthInKbFileWithRandomOrFixedCharacters(int sizeInKb, boolean isRandom) throws IOException {
```

Comment:
It seems we can use RandomTempFile https://github.com/aws/aws-sdk-java-v2/blob/master/test/test-utils/src/main/java/software/amazon/awssdk/testutils/RandomTempFile.java